astroid/nodes/node_classes.py

```diff
@@ -2578,44 +2578,24 @@ class If(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
     _astroid_fields = ("test", "body", "orelse")
     _multi_line_block_fields = ("body", "orelse")
 
-    def __init__(
-        self,
-        lineno: int,
-        col_offset: int,
-        parent: NodeNG,
-        *,
-        end_lineno: int | None,
-        end_col_offset: int | None,
-    ) -> None:
-        self.test: NodeNG
-        """The condition that the statement tests.
-
-        This attribute gets set in the postinit method.
-        """
-
-        self.body: list[NodeNG] = []
-        """The contents of the block."""
-
-        self.orelse: list[NodeNG] = []
-        """The contents of the ``else`` block."""
+    test: NodeNG
+    """The condition that the statement tests."""
 
-        self.is_orelse: bool = False
-        """Whether the if-statement is the orelse-block of another if statement."""
+    body: list[NodeNG]
+    """The contents of the block."""
 
-        super().__init__(
-            lineno=lineno,
-            col_offset=col_offset,
-            end_lineno=end_lineno,
-            end_col_offset=end_col_offset,
-            parent=parent,
-        )
+    orelse: list[NodeNG]
+    """The contents of the ``else`` block."""
 
     def postinit(self, test: NodeNG, body: list[NodeNG], orelse: list[NodeNG]) -> None:
         self.test = test
         self.body = body
         self.orelse = orelse
-        if isinstance(self.parent, If) and self in self.parent.orelse:
-            self.is_orelse = True
+
+    @cached_property
```

Comment:
I don't like how the `is_orelse` field had a default value which was sometimes overwritten in `postinit`. So I changed it to have no default value and calculated the value in `postinit`. But then that raised some kind of circularity issue at build time. So I changed it to a `cached_property`, which is equivalent and I think ultimately more sensible.

Reply:
Is this condition ever True? I think I tested this and the condition never passed in our test suite. It might be redundant now.

Reply:
It's used in just one place that I'm aware of, in `_filter_stmts`:

```python
                # If the if statement is first-level and not within an orelse block
                # we know that it will be evaluated
                elif not if_parent.is_orelse:
                    _stmts = [node]
                    _stmt_parents = [stmt.parent]
```

Reply:
I added an exception to the else-block and it seems to never get reached:

```python
                # If the if statement is first-level and not within an orelse block
                # we know that it will be evaluated
                elif not if_parent.is_orelse:
                    _stmts = [node]
                    _stmt_parents = [stmt.parent]
                # Else we do not known enough about the control flow to be 100% certain
                # and we append to possible statements
                else:
                    raise Exception  # <-- never reached
                    _stmts.append(node)
                    _stmt_parents.append(stmt.parent)
```

So it seems the whole `is_orelse` field can be cut.

Reply:
Let's remove it then!